web/src/topic_list.js

```diff
@@ -314,21 +314,10 @@ export function initialize({narrow_on_topic_click}) {
             return;
         }
 
-        // In a more componentized world, we would delegate some
-        // of this stuff back up to our parents.
-
         const $stream_row = $(e.target).parents(".narrow-filter");
         const stream_id = Number.parseInt($stream_row.attr("data-stream-id"), 10);
-        const sub = sub_store.get(stream_id);
         const topic = $(e.target).parents("li").attr("data-topic-name");
-
-        narrow_on_topic_click(
-            [
-                {operator: "stream", operand: sub.name},
-                {operator: "topic", operand: topic},
-            ],
-            {trigger: "sidebar"},
-        );
+        on_topic_click(stream_id, topic);
```

Comment:
(If this API cleanup makes sense to you, we can squash it into the earlier commit. Likewise for stream_list.)

Reply:
I think this is OK but I'd maybe still call it `narrow_on_topic_click` -- just to provide the hint to someone reading this function what it's going to do without having to go find where this gets called.

Reply:
Idunno, maybe it's worth a #frontend conversation about these naming conventions and we can change it or not based on what folks think about that.